src/main/java/page/ProductPage.java

```diff
@@ -0,0 +1,98 @@
+package page;
+import java.util.concurrent.TimeUnit;
+
+import java.util.ArrayList;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.PageFactory;
+
+import utilities.BaseClass;
+
+public class ProductPage extends BaseClass {
+	
+	private WebDriver driver;
+	int Rindex, Sindex;
+	int[] Ratings = new int[40];
+	String[] StarRating = new String[40];
+	String[] ProductDetail = new String[40];
+	int Index = 0, Heigest = 0;
+	ArrayList<WebElement> xList = new ArrayList<WebElement>();
+	
+	public ProductPage(WebDriver driver) {
+		super(driver);
+		this.driver=driver;
+		PageFactory.initElements(driver, this);
+	}
+
+	public void view() {
+		driver.manage().timeouts().implicitlyWait(20, TimeUnit.SECONDS);
+		if (driver.findElements(By.xpath("//div[@data-id]")).size() == 24) {
```

Comment:
find another way to distinguish between list and grid view, this is not a correct way !!